README.md

```diff
@@ -1,17 +1,28 @@
-# Socket Component
+# react/socket
```

Comment:
Thanks @jsor and @WyriHaximus, I think you both made a very valid point and it doesn't make much sense to change this as part of this PR in the first place :+1: 

I've updated this to just "Socket" (originally "Socket Component"), because the paragraph below uses the words "server and client component" :shipit: